solutions/usaco-1037.mdx

```diff
@@ -0,0 +1,136 @@
+---
+id: usaco-1037
+source: USACO Bronze 2020 US Open
+title: Cowntact Tracing
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://usaco.org/current/data/sol_tracing_bronze_open20.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N \cdot T^2)$
+
+<LanguageSection>
+
+<JavaSection>
+ 
+```java
+import java.util.*;
+import java.io.*;
+
+public class CowntactTracing {
+	static boolean[] cowEndsInfected = new boolean[101];
+	static int n;
+	static int[] cowX = new int[251];
+	static int[] cowY = new int[251];
```

Comment:
arrays shouldn't be constant size for java, there are some below as well

Reply:
so are constant size arrays fine for c++? the official solution used one, which is why I did that

Reply:
yes, it is fine for c++, but good practice to use variable length arrays for java.

Reply:
c++ needs to have constant sized arrays (variable length aren't c++ standard), but java can (and should? i'm not familiar w/ java) have variable length arrays

Reply:
yea c++ standard doesn't support variable array lengths.

Reply:
i can only change one of them to be not constant sized, the rest need to be constant sized because it is needed to solve the problem

Reply:
i don't remember having the use constant size arrays to solve this problem?

Reply:
implementation difference then

Reply:
ok well i see the 250 alot, it'd be really cool if you could put it into a variable like `MAX_N`